src/Microsoft.AspNetCore.Mvc.Core/ProblemDetails.cs

```diff
@@ -11,34 +12,57 @@ namespace Microsoft.AspNetCore.Mvc
     /// </summary>
     public class ProblemDetails
     {
+        /// <summary>
+        /// Gets or sets values for extension members. Extension members appear in the same namespace as
+        /// other members of a problem type.
+        /// </summary>
+        /// <param name="key">The extension member name.</param>
+        /// <returns>The value.</returns>
+        public object this[string key]
```

Comment:
Consider restricting this and `ExtensionMembers` to `string` values. Json.NET will often read back `JObject`s or `JArray`s rather than the original types and XML de-serializers should do something similar.

Reply:
The spec actually allows for arbitrary objects and it feels less than nice to remove this ability because we don't round trip correctly. Think it would make sense to add a note that says this is slightly problematic for complex types in XML (XML does not like arbitrary complex types in there and JSON may not accurately round trip this)